commands/operator-sdk/cmd/new.go

```diff
@@ -203,3 +310,66 @@ func initGit() {
 	execCmd(os.Stdout, "git", "commit", "-q", "-m", "INITIAL COMMIT")
 	fmt.Fprintln(os.Stdout, "Run git init done")
 }
+
+// Copied from add/api.go command
+func updateRoleForResource(r *scaffold.Resource, absProjectPath string) error {
+	// append rbac rule to deploy/role.yaml
+	roleFilePath := filepath.Join(absProjectPath, "deploy", "role.yaml")
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	obj, _, err := cgoscheme.Codecs.UniversalDeserializer().Decode(roleYAML, nil, nil)
+	if err != nil {
+		return fmt.Errorf("failed to decode role manifest %v: %v", roleFilePath, err)
+	}
+	switch role := obj.(type) {
+	// TODO: use rbac/v1.
```

Comment:
This TODO is completed

Reply:
Shawn I know you were thinking about the possibility of generating and handling ClusterRoles in the operator(along with watching all namesapces).
For now we update the existing role.yaml with a new rule whenever add a new API `operator-sdk add api`.
```Go
case *rbacv1.Role:
...
default:
	return errors.New("failed to parse role.yaml as a role")
```
Can you change this TODO to handle ClusterRole.